File: algoritmos/diffie-hellman/dh.js
const public_hash = (o) => {
    return (o.g ** o.s) % o.n;
};

let alice = {
    g: 17,
    n: 648123,
    s: 325
};

console.log(alice.g ** alice.s);

console.log(alice, public_hash(alice));

File: algoritmos/diffie-hellman/dh.py
import random
import gmpy2

GEN = 7
BASE = 975857477387853487

class author:
    def __init__(self):
        self.secret = random.randint(1, BASE)

    def public_hash(self):
        return (GEN ** self.secret) % BASE
    
    def shared_hash(self, hash):
        return (hash ** self.secret) % BASE

alice = author()
bob = author()

print(alice.secret)
print(alice.public_hash())
print(bob.secret)
print(bob.public_hash())
print("")
print(alice.shared_hash(bob.public_hash()))
print(bob.shared_hash(alice.public_hash()))

